lib/iris/fileformats/netcdf.py

```diff
@@ -1812,6 +1810,8 @@ def _get_coord_variable_name(self, cube, coord):
         else:
             name = coord.standard_name or coord.long_name
             if not name or set(name).intersection(string.whitespace):
+                # NB don't know how to fix this if it is not a Coord or similar
+                assert isinstance(coord, iris.coords.Coord)
```

Comment:
Because we are about to use 'coord_dims'.
We could turn that around and use 'element.cube_dims(cube)', but this still doesn't work for Connectivities
Because Connectivities, of course, **aren't** mapped to cube dims.

In fact, anything which is a _DimensionalMetadata [has a cube_dims](https://github.com/SciTools/iris/blob/6c2dc49122d4c862e4d6a3e16bd0eeb4c49d09b0/lib/iris/coords.py#L185-L199), 
but we made a (maybe slightly nasty) exception by [disabling the cube_dims function on Connectivities](https://github.com/SciTools/iris/blob/6c2dc49122d4c862e4d6a3e16bd0eeb4c49d09b0/lib/iris/experimental/ugrid/__init__.py#L628-L630)

Reply:
TBH I'm not the biggest fan of the `cube_dims` method, and it exposes all sorts of questionable behavioural awkwardness/weirdness for `Connectivity`... so it would be kinda nice not to build on that; we'll see I guess.